src/bytes_mut.rs

```diff
@@ -422,11 +422,9 @@ impl BytesMut {
     /// assert_eq!(buf, b"hello"[..]);
     /// ```
     pub fn truncate(&mut self, len: usize) {
-        if len < self.len() {
-            unsafe {
-                // SAFETY: Shrinking the buffer cannot expose uninitialized bytes.
-                self.set_len(len);
-            }
+        if len <= self.len() {
```

Comment:
Taking https://github.com/rust-lang/rust/issues/124346 and modifying it a bit the `<=` doesn't seem to help https://rust.godbolt.org/z/K1c437eYj